mava/systems/tf/mappo/training.py

```diff
@@ -371,37 +383,41 @@ def _forward(self, inputs: Any) -> None:
 
                 # Generalized Advantage Estimation
                 gae = tf.stop_gradient(td_lambda_extra.temporal_differences)
-                mean, variance = tf.nn.moments(gae, axes=[0, 1], keepdims=True)
-                normalized_gae = (gae - mean) / tf.sqrt(variance)
-
-                policy_gradient_loss = tf.reduce_mean(
-                    -tf.minimum(
-                        tf.multiply(importance_ratio, normalized_gae),
-                        tf.multiply(clipped_importance_ratio, normalized_gae),
-                    ),
+
+                # Note (dries): Maybe add this in again? But this might be breaking
+                # training. mean, variance = tf.nn.moments(gae, axes=[0, 1],
+                # keepdims=True)
+                # normalized_gae = (gae - mean) / tf.sqrt(variance)
```

Comment:
Was there evidence this was breaking training / hurting performance? Maybe we just need to make sure the denominator is not 0 , by doing this ` normalized_gae = (gae - mean) / (tf.sqrt(variance)+ 1e-8)` .

Reply:
I would have to investigate this again. I just remembered that this caused NaNs. I cannot remember if I tried adding a small constant below the division. It's definitely a good idea yes:+1: